### PR TITLE
Nondet trait

### PR DESCRIPTION
**Pull request by alastairreid**
Note: the file to read first is the last file: verification-annotations/src/traits.rs
This shows the new API that all the other changes are implementing/using

### CODE REVIEW

File: verification-annotations/src/crux.rs [hunk 1]
```diff
@@ -10,17 +10,31 @@
...
+    fn abstract_value() -> Self {
+        // We assume the string argument is just for reporting to the user, and
+        // doesn't affect the results.
+        let r : T = crucible::Symbolic::symbolic("");
+        r
+    }
 }
 
+impl <T: crucible::Symbolic + Default> Symbolic for T {
+    fn symbolic(_desc: &'static str) -> Self {
```
Comment (fshaked) [inline verification-annotations/src/crux.rs]: Maybe pass str to symbolic below, instead of ""?

Comment (alastairreid) [reply verification-annotations/src/crux.rs]: oops! Good catch.

File: verification-annotations/src/klee.rs [hunk 2]
```diff
@@ -105,8 +147,8 @@ pub fn expect(msg: Option<&str>) {
 macro_rules! assert {
     ($cond:expr) => {
         if ! $cond {
-            eprintln!("VERIFIER: assertion failed: {}", stringify!($cond));
-            abort();
+            eprintln!("VERIFIER: panicked at 'assertion failed: {}', {}:{}:{}", stringify!($cond), std::file!(), std::line!(), std::column!());
```
Comment (fshaked) [inline verification-annotations/src/klee.rs:150]: I'm not sure "panicked" fits here

Comment (alastairreid) [reply verification-annotations/src/klee.rs:150]: This is a change to make verifier::assert work the same as std::assert for KLEE.
In particular, to make should_panic work, it needs to have "panicked at" in the message.

Another way to implement this change is to have this macro call 'crate::panic' and then 'crate::panic' would print the 'panicked at' part of the string

Comment (fshaked) [reply verification-annotations/src/klee.rs:150]: Calling crate::panic makes it worse. I still think using "panicked" is not nice, but I guess nice is not our top priority.

**Review by fshaked — APPROVED** [review]

**Pull request merged**